src/main/java/jenkins/plugin/randomjobbuilder/LoadGeneration.java

```diff
@@ -0,0 +1,242 @@
+package jenkins.plugin.randomjobbuilder;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Action;
+import hudson.model.Cause;
+import hudson.model.CauseAction;
+import hudson.model.Descriptor;
+import hudson.model.Job;
+import hudson.model.Queue;
+import hudson.model.Run;
+import hudson.model.queue.QueueTaskFuture;
+import hudson.security.ACL;
+import hudson.util.DescribableList;
+import jenkins.model.Jenkins;
+import jenkins.model.ParameterizedJobMixIn;
+import net.sf.json.JSONObject;
+import org.acegisecurity.context.SecurityContext;
+import org.acegisecurity.context.SecurityContextHolder;
+import org.kohsuke.stapler.StaplerRequest;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * Configurable load generator that runs jobs
+ */
+public class LoadGeneration extends AbstractDescribableImpl<LoadGeneration>  {
```

Comment:
Mostly a bag of general small things needed to hook the bits together (i.e. mini-classes, utility functions, and the top-level describable/descriptor)